cardano-cli/src/Cardano/CLI/Types.hs

```diff
@@ -306,9 +306,12 @@ data ScriptWitnessFiles witctx where
 
 deriving instance Show (ScriptWitnessFiles witctx)
 
+-- TODO: Do we want the ability to specify datums for non-spending scripts?
+-- Currently we can't (although its not necessary)
```

Comment:
Usually we use the term `datum` for the information which is only passed to `SPEND` scripts (ie *not* certs, withdrawals, or minting). This is the information inside the transaction output (either a hash or an inline datum). So if I am understanding the question above correctly, the answer is no, only spending scripts get datums.